File: ployglots/mta-ployglot-pkg-fetch-client/packages/py/examples/integrations/jira.py
"""Example: JIRA REST API v3.

Env:
    JIRA_HOST   — e.g. 'https://your-domain.atlassian.net'
    JIRA_USER   — your Atlassian email
    JIRA_PASS   — a JIRA API token
                  https://id.atlassian.com/manage-profile/security/api-tokens
    HTTPS_PROXY — optional outbound proxy, picked up by build_proxy({}).

Endpoint exercised: GET /rest/api/3/myself — returns the caller's profile.

Run:
    python -m examples.integrations.jira
"""

from __future__ import annotations

import asyncio
import sys

from fetch_http_client import AsyncClient, BasicAuth

from .._shared import build_proxy, require_env


async def main() -> None:
    host = require_env("JIRA_HOST")
    user = require_env("JIRA_USER")
    password = require_env("JIRA_PASS")

    # proxy={} → auto-detect from HTTPS_PROXY / HTTP_PROXY.
    # Pass {"host": ..., "user": ..., "pass": ...} to override explicitly.
    proxy = build_proxy({})

    kwargs: dict = {
        "base_url": host,
        "auth": BasicAuth(user, password),
        "headers": {"accept": "application/json"},
    }
    if proxy is not None:
        kwargs["proxy"] = proxy

    async with AsyncClient(**kwargs) as client:
        resp = await client.get("/rest/api/3/myself")
        resp.raise_for_status()
        me = resp.json()
        sys.stdout.write(
            f"JIRA {host} → {me.get('displayName')} "
            f"<{me.get('emailAddress')}> ({me.get('accountId')})\n"
        )


if __name__ == "__main__":
    try:
        asyncio.run(main())
    except Exception as err:
        sys.stderr.write(f"JIRA example failed: {err}\n")
        sys.exit(1)

File: ployglots/mta-ployglot-pkg-fetch-client/packages/ts/examples/integrations/jira.ts
// @ts-nocheck
/**
 * Example: JIRA REST API v3.
 *
 * Env:
 *   JIRA_HOST  — e.g. 'https://your-domain.atlassian.net'
 *   JIRA_USER  — your Atlassian email
 *   JIRA_PASS  — a JIRA API token
 *     (https://id.atlassian.com/manage-profile/security/api-tokens)
 *   HTTPS_PROXY — optional outbound proxy, picked up by buildProxy({}).
 *
 * Endpoint exercised: GET /rest/api/3/myself — returns the caller's profile.
 *
 * Run: tsx examples/integrations/jira.ts
 */
import { AsyncClient, BasicAuth } from '@polyglot/fetch-http-client';
import { buildProxy, requireEnv } from '../_shared.js';

async function main(): Promise<void> {
  const host = requireEnv('JIRA_HOST');
  const user = requireEnv('JIRA_USER');
  const pass = requireEnv('JIRA_PASS');

  // proxy = {} → auto-detect from HTTPS_PROXY / HTTP_PROXY.
  // Pass { host, user, pass } to override explicitly.
  const proxy = buildProxy({});

  const client = new AsyncClient({
    baseUrl: host,
    auth: new BasicAuth(user, pass),
    headers: { accept: 'application/json' },
    ...(proxy ? { proxy } : {}),
  });

  try {
    const resp = await client.get('/rest/api/3/myself');
    resp.raiseForStatus();
    const me = await resp.json<{
      accountId: string;
      displayName: string;
      emailAddress: string;
    }>();
    process.stdout.write(
      `JIRA ${host} → ${me.displayName} <${me.emailAddress}> (${me.accountId})\n`,
    );
  } finally {
    await client.close();
  }
}

main().catch((err: unknown) => {
  process.stderr.write(`JIRA example failed: ${(err as Error).message}\n`);
  process.exitCode = 1;
});
